tests/functional/adapter/test_hive_on_schema_change.py

```diff
@@ -0,0 +1,68 @@
+import pytest
+
+from dbt.contracts.results import RunStatus
+from dbt.tests.util import run_dbt, run_dbt_and_capture
+
+models__table_base_model = """
```

Comment:
great test, how about replicating it for iceberg tables as well, just to be sure that we don't break anything in there?

Reply:
I parameterized the tests so that they now run once with `table_type=hive` and once with `table_type=iceberg`. The file name was "test_hive_on_schema_change" and I renamed it to "test_on_schema_change" since it covers both table types.